### PR TITLE
fix: Propagate `lr_scheduler_kwargs` options to create LR Scheduler when LayerWiseDummyOptimizer is used

### PR DESCRIPTION
**Pull request by BlackNoodle**
# What does this PR do?

Currently, when using `LayerWiseDummyOptimizer` (e.g., GaLore Optimizers), LR schedulers are not properly initialized. For example, attempting to use `cosine_with_min_lr` with `min_lr_rate = 0.1` results in an error. To fix this, the `scheduler_specific_kwargs` argument should also be passed during the recursive call of `get_scheduler`.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by muellerzr — APPROVED** [review]
Thanks, I agree this solution makes sense.

**Review by DaizeDong — APPROVED** [review]

**Comment by DaizeDong** [comment]
I recently encountered the same issue, and this fix is essential. Please merge it.

**Comment by Rocketknight1** [comment]
cc @sunmarc @mekkcyber are you okay with this being merged?

**Review by SunMarc — APPROVED** [review]
Thx ! crazy that we missed this one for so long
